# Cleaning and Feature Selection

Objective: load the merged player+stats dataset from notebook 04, drop
redundant/irrelevant columns, and prepare a clean feature set for the
Phase 1 cost-benefit scoring and clustering steps.

In [11]:
import pandas as pd

## Loading merged data

In [12]:
df_pws = pd.read_csv('../data/processed/players_with_stats_2024.csv')

## Dropping redundant and irrelevant columns

In [13]:
columns_to_drop = [
    # duplicate/redundant merge artifacts
    'name_y', 'name_normalized', 'stats_key', 'name_normalized_stats',
    
    # irrelevant metadata
    'url_x', 'url_y', 'image_url', 'filename', 'agent_name', 'player_code',
    
    # club-level info, redundant with current_club_name (one row per player already)
    'club_id', 'club_code', 'domestic_competition_id', 'total_market_value',
    'squad_size', 'average_age', 'foreigners_number', 'foreigners_percentage',
    'national_team_players', 'stadium_name', 'stadium_seats',
    'net_transfer_record', 'coach_name', 'last_season_y',
    
    # matching process helpers, no longer needed
    'matched', 'used', 'last_season_x',
]

df_clean = df_pws.drop(columns=columns_to_drop)
df_clean.shape

(1021, 39)

In [14]:
df_clean.columns.tolist()

['player_id',
 'first_name',
 'last_name',
 'name_x',
 'current_club_id',
 'country_of_birth',
 'city_of_birth',
 'country_of_citizenship',
 'date_of_birth',
 'sub_position',
 'position',
 'foot',
 'height_in_cm',
 'contract_expiration_date',
 'international_caps',
 'international_goals',
 'current_national_team_id',
 'current_club_domestic_competition_id',
 'current_club_name',
 'market_value_in_eur',
 'highest_market_value_in_eur',
 'Jogador',
 'Pos.',
 'Time',
 'Min.',
 'Gols',
 'Assis.',
 'xG',
 'xAG',
 'PrgP',
 'PrgC',
 'Tent',
 'Suc',
 'SCA',
 'GCA',
 'Div',
 'Bloqueios',
 'CrtsA',
 'CrtV']

## Standardizing column names

In [15]:
df_clean = df_clean.rename(columns={
    'name_x': 'player_name',
    'Jogador': 'stats_player_name',  # redundant with player_name where matched, kept for traceability
    'Pos.': 'stats_position',
    'Time': 'stats_club',
    'Min.': 'minutes_played',
    'Gols': 'goals',
    'Assis.': 'assists',
    'PrgP': 'progressive_passes',
    'PrgC': 'progressive_carries',
    'Tent': 'dribbles_attempted',
    'Suc': 'dribbles_successful',
    'SCA': 'shot_creating_actions',
    'GCA': 'goal_creating_actions',
    'Div': 'tackles',
    'Bloqueios': 'blocks',
    'CrtsA': 'yellow_cards_stats',
    'CrtV': 'red_cards_stats',
})
df_clean.columns.tolist()

['player_id',
 'first_name',
 'last_name',
 'player_name',
 'current_club_id',
 'country_of_birth',
 'city_of_birth',
 'country_of_citizenship',
 'date_of_birth',
 'sub_position',
 'position',
 'foot',
 'height_in_cm',
 'contract_expiration_date',
 'international_caps',
 'international_goals',
 'current_national_team_id',
 'current_club_domestic_competition_id',
 'current_club_name',
 'market_value_in_eur',
 'highest_market_value_in_eur',
 'stats_player_name',
 'stats_position',
 'stats_club',
 'minutes_played',
 'goals',
 'assists',
 'xG',
 'xAG',
 'progressive_passes',
 'progressive_carries',
 'dribbles_attempted',
 'dribbles_successful',
 'shot_creating_actions',
 'goal_creating_actions',
 'tackles',
 'blocks',
 'yellow_cards_stats',
 'red_cards_stats']

## Exporting cleaned dataset

In [16]:
df_clean.to_csv('../data/processed/players_final_2024.csv', index=False)

## Conclusion

- Loaded the merged roster+stats dataset from notebook 04 (1021 players, 66 columns)
- Dropped 25 redundant/irrelevant columns (merge artifacts, club-level
  metadata duplicated across rows, matching-process helper columns),
  resulting in 40 columns
- Standardized column names to English/snake_case for consistency with
  the rest of the project
- Result exported to data/processed/players_final_2024.csv, ready for
  Phase 1 cost-benefit scoring and clustering